--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -4,6 +4,33 @@ module(
     repo_name = "io_grpc_grpc_java",
     version = "1.78.0.bcr.1",  # CURRENT_GRPC_VERSION
 )
+
+http_file = use_repo_rule("@bazel_tools//tools/build_defs/repo:http.bzl", "http_file")
+
+http_file(
+    name = "grpc_java_plugin_linux_aarch_64",
+    executable = True,
+    sha256 = "c95caf0ab8a27e01e53f8cff8e8947f6a8841afaa357fa4b0679b4af45731886",
+    urls = ["https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.78.0/protoc-gen-grpc-java-1.78.0-linux-aarch_64.exe"],
+)
+http_file(
+    name = "grpc_java_plugin_linux_x86_64",
+    executable = True,
+    sha256 = "c02dd4d757b6794072a1675c2a9f232631b64c3d53c1ec19effb0504859a82ff",
+    urls = ["https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.78.0/protoc-gen-grpc-java-1.78.0-linux-x86_64.exe"],
+)
+http_file(
+    name = "grpc_java_plugin_osx_aarch_64",
+    executable = True,
+    sha256 = "b0722795401320a71130395f1a3d8a6534b7311008cce8d78f8cb7ce176daa26",
+    urls = ["https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.78.0/protoc-gen-grpc-java-1.78.0-osx-aarch_64.exe"],
+)
+http_file(
+    name = "grpc_java_plugin_osx_x86_64",
+    executable = True,
+    sha256 = "b0722795401320a71130395f1a3d8a6534b7311008cce8d78f8cb7ce176daa26",
+    urls = ["https://repo1.maven.org/maven2/io/grpc/protoc-gen-grpc-java/1.78.0/protoc-gen-grpc-java-1.78.0-osx-x86_64.exe"],
+)
 
 # GRPC_DEPS_START
 IO_GRPC_GRPC_JAVA_ARTIFACTS = [
